include/oneapi/dpl/pstl/algorithm_impl.h

```diff
@@ -2452,6 +2452,37 @@ __pattern_stable_sort(__parallel_tag<_IsVector>, _ExecutionPolicy&& __exec, _Ran
     });
 }
 
+// Add separate patterns for std::ranges::sort due to std::indirectly_swappable requirement,
+// which implies the use of std::ranges::iter_swap, which can be customized externally
+#if _ONEDPL_CPP20_RANGES_PRESENT
+template <class _Tag, class _ExecutionPolicy, class _RandomAccessIterator, class _Compare>
+void
+__pattern_stable_sort_ranges(_Tag, _ExecutionPolicy&&, _RandomAccessIterator __first, _RandomAccessIterator __last,
```

Comment:
I would suggest not introducing another pattern for sort, just add LeafSort type to existed __pattern_stable_sort instead.

Reply:
Yep, I will try it

Reply:
I've implemented it in #1843. If you find it useful, then lets continue work there and close this PR.